projects/vdk-core/src/vdk/internal/builtin_plugins/run/file_based_step.py

```diff
@@ -105,6 +106,15 @@ def run(job_input)
                     finally:
                         if success:
                             log.info("Exiting  %s#run(...) SUCCESS" % filename)
+                            user_errors_in_step_handled = [
+                                i
+                                for i in errors.resolvable_context().resolvables.get(
+                                    ResolvableByActual.USER, []
+                                )
+                                if str(step.file_path) == i.caused_by_step_file
+                            ]
+                            for e in user_errors_in_step_handled:
+                                e.resolved = True
                         else:
```

Comment:
The whole "resolved" logic based on step file seems error prone. Note that in the future we can have steps that aren to produced out of files. I actually have a POC on my computer for SQL splitter which splits SQL file into multiple steps using delimiter. 


Why not do "errors.resolvable_context().clear()" on success. Success means the errors from teh stpe are not relevant and can be cleared ?

Reply:
We  synced face to face we agree that we should keep the history of the errors (for troubleshooting purposes) but 
A) we should mark all errors as resolved up to this point (regardless of step or user/platform error)